presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/StandardColumnMappings.java

```diff
@@ -369,4 +399,87 @@ public static LongWriteFunction timestampWriteFunction()
         }
         return Optional.empty();
     }
+
+    public static Object jdbcObjectToPrestoNative(Object jdbcObject, Type prestoType)
+    {
+        if (jdbcObject == null) {
+            return null;
+        }
+        else if (prestoType instanceof DecimalType) {
+            DecimalType decimalType = (DecimalType) prestoType;
+            BigDecimal value = (BigDecimal) jdbcObject;
+            if (decimalType.isShort()) {
+                return encodeShortScaledValue(value, decimalType.getScale());
+            }
+            else {
+                return encodeScaledValue(value, decimalType.getScale());
+            }
+        }
+        else if (REAL.equals(prestoType)) {
+            return floatToRawIntBits((float) jdbcObject);
+        }
+        else if (jdbcObject instanceof Date) {
+            return toLong((Date) jdbcObject);
+        }
+        else if (jdbcObject instanceof Time) {
+            return toLong((Time) jdbcObject);
+        }
+        else if (jdbcObject instanceof Timestamp) {
+            return ((Timestamp) jdbcObject).getTime();
+        }
+        else {
+            return jdbcObject;
+        }
+    }
+
+    public static Object prestoNativeToJdbcObject(Type prestoType, Object prestoNative)
+    {
+        if (prestoNative == null) {
+            return null;
+        }
+        else if (prestoType instanceof DecimalType) {
+            DecimalType decimalType = (DecimalType) prestoType;
+            if (decimalType.isShort()) {
+                return toBigDecimal(decimalType, (long) prestoNative);
+            }
+            else {
+                return toBigDecimal(decimalType, (Slice) prestoNative);
+            }
+        }
+        else if (REAL.equals(prestoType)) {
+            return intBitsToFloat(toIntExact((long) prestoNative));
+        }
+        else if (TINYINT.equals(prestoType)) {
+            return SignedBytes.checkedCast((long) prestoNative);
+        }
+        else if (SMALLINT.equals(prestoType)) {
+            return Shorts.checkedCast((long) prestoNative);
+        }
+        else if (INTEGER.equals(prestoType)) {
+            return toIntExact((long) prestoNative);
+        }
+        else if (DATE.equals(prestoType)) {
+            return toDate((long) prestoNative);
+        }
+        else if (TIME.equals(prestoType)) {
+            return new Time((long) prestoNative);
+        }
+        else if (TIMESTAMP.equals(prestoType)) {
+            return new Timestamp((long) prestoNative);
```

Comment:
That's not correct. With legacy timestamp semantics (default), `prestoNative` is a utc millis (Java timestamp) which interpreted in *session* zone yields year/month/day/hour/minutes that the value represents.
OTOH, `java.sql.TImestamp(x)` treats `x` as utc millis (Java timestamp) which interpreted in *JVM* zone yields year/month/day/hour/minutes that the value represents.
To fix this you would need:
- write different conversion code
- pass `ConnectorSession` so that the code can do condition on `io.prestosql.spi.connector.ConnectorSession#isLegacyTimestamp`

... and even then the code would not be generic. `java.sql.Timestamp` cannot represent certain values and many JDBC drivers (but not all) support `java.time.LocalDateTime` and we should use that.

For this reasons, the `prestoNativeToJdbcObject` cannot be done in a generic way, without connector-specific code. I would therefore suggest two things:

1. move this conversion to PostgresClient (or some companion helper utility class -- if possible)
   * then make sure there are no unnecessary changes in `StandardColumnMappings.java` (i would expect none are needed)
2. pass `ConnectorSession` here
3. remove support for time and timestamp types from this PR. It's easy to get them wrong and it's better that we focus on more generic stuff first and add timestamp arrays later.
   * to ensure we (and others) do not forget about timesamps and the necessary abstractions they require, I would keep the `TIMESTAMP.equals(prestoType)` condition like:
     ```
     if (TIMESTAMP.equals(prestoType)) {
         throw new PrestoException(.....); // TODO
     ```